frb_dart/lib/src/ffi/io.dart

```diff
@@ -51,26 +37,134 @@ class DartApiDl {
 /// This class is only for internal usage.
 class WireSyncReturnStruct extends ffi.Struct {
   /// Not to be used by normal users, but has to be public for generated code
-  external ffi.Pointer<ffi.Uint8> ptr;
+  external DartCObject data;
 
-  /// Not to be used by normal users, but has to be public for generated code
-  @ffi.Int32()
-  external int len;
+  @ffi.Bool()
+  external bool isSuccess;
 
-  /// Not to be used by normal users, but has to be public for generated code
-  @ffi.Uint8()
-  external int success;
+  external ffi.Pointer<ffi.Void> ptr;
```

Comment:
Yes, it is used in `WireSyncReturnStruct::free`.

`WireSyncReturnStruct` copies values instead of wrapping a pointer, which is slighly more simple and efficient (right?). Keeping the pointer to the wrapped data allows to `ManuallyDrop` later (`free_WireSyncReturnStruct`) thus letting `allo_isolate`'s `Drop` impl correctly drop the returned struct.

Do you feel we should use `ptr` directly to access value and not copied values?